corehq/motech/fhir/tasks.py

```diff
@@ -122,10 +123,10 @@ def claim_service_request(requests, service_request, case_id):
     headers = {'If-Match': etag}
     response = requests.put(endpoint, json=service_request, headers=headers)
     if 200 <= response.status < 300:
-        return
+        return service_request
     if response.status == 412:
         # ETag didn't match. Try again.
-        claim_service_request(requests, service_request, case_id)
+        return claim_service_request(requests, service_request, case_id)
```

Comment:
That seems really unlikely. [412 is meant for ETags](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/412), so looping here would mean the remote system is changing the ETag faster than we can query it ... or is broken? ... or is malicious? (in which case, waiting for an incoming request so that you can send a 412 response feels like an ineffective attack strategy.)

What do you think? Add a counter?

Reply:
(I added a counter.)